src/librbd/ImageCtx.h

```diff
@@ -122,6 +122,9 @@ namespace librbd {
     librados::IoCtx data_ctx;
     librados::IoCtx md_ctx;
 
+    // optional rados cluster owned by this image
+    std::unique_ptr<librados::Rados> rados;
```

Comment:
I'm getting this error when replacing the `unique_ptr` with a `librados::Rados`:

```
/home/oro/ceph/src/librbd/migration/NativeFormat.cc:112:34: error: 'const librados::v14_2_0::Rados& librados::v14_2_0::Rados::operator=(const librados::v14_2_0::Rados&)' is private within this context
  112 |     m_image_ctx->migration_rados = rados;
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
In file included from /home/oro/ceph/src/librbd/deep_copy/Types.h:8,
                 from /home/oro/ceph/src/librbd/Types.h:9,
                 from /home/oro/ceph/src/librbd/migration/NativeFormat.h:8,
                 from /home/oro/ceph/src/librbd/migration/NativeFormat.cc:4:
/home/oro/ceph/src/include/rados/librados.hpp:1560:18: note: declared private here
 1560 |     const Rados& operator=(const Rados& rhs);
      |                  ^~~~~~~~
```

Reply:
> This is expected because `Rados` is a handle type which can't be copied, just like `std::unique_ptr`. You probably want `m_image_ctx->migration_rados = std::move(rados)`.

Tried this, but getting the exact same error:

```
/home/oro/ceph/src/librbd/migration/NativeFormat.cc:112:34: error: 'const librados::v14_2_0::Rados& librados::v14_2_0::Rados::operator=(const librados::v14_2_0::Rados&)' is private within this context
  112 |     m_image_ctx->migration_rados = std::move(rados);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
In file included from /home/oro/ceph/src/librbd/deep_copy/Types.h:8,
                 from /home/oro/ceph/src/librbd/Types.h:9,
                 from /home/oro/ceph/src/librbd/migration/NativeFormat.h:8,
                 from /home/oro/ceph/src/librbd/migration/NativeFormat.cc:4:
/home/oro/ceph/src/include/rados/librados.hpp:1560:18: note: declared private here
 1560 |     const Rados& operator=(const Rados& rhs);
      |                  ^~~~~~~~
```

Reply:
Apologies, it looks like `librados::Rados` doesn't have move assignment defined, only `neorados::RADOS` does.  I confused the two in my head.

Adding a move constructor and assignment operator should be easy but out of scope here.  Feel free to proceed with `std::unique_ptr`.